.github/workflows/build-and-push-cloud-qcow2.yml

```diff
@@ -11,13 +11,12 @@ env:
   # TODO use quay once it supports artifacts
   DISKIMAGE_REGISTRY: ghcr.io
 
-permissions:
-  contents: read
-  packages: write
-
 jobs:
   build-and-push-cloud-qcow2-image:
     runs-on: ubuntu-latest
+    permissions:
```

Comment:
Thanks for fixing this!  I'm a bit confused though...the syntax of having the permissions at the toplevel is something I've used elsewhere and is certainly documented to work, xref https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

Reply:
Both options are valid. I think that having it on the job level is more secure. It impacts this job only in case another job is added to this workflow.